linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5677,28 +5691,35 @@ private[optimizer] object OptimizerCore {
       RecordType(recordFields)
     }
 
-    private val recordFieldNames: Map[FieldName, RecordType.Field] = {
+    private val recordFields: Map[FieldName, RecordType.Field] = {
       val elems = for ((fieldDef, recordField) <- allFields.zip(recordType.fields))
         yield fieldDef.name.name -> recordField
       elems.toMap
     }
 
+    private[OptimizerCore] def findField(fieldName: FieldName): RecordType.Field =
+      recordFields(fieldName)
+
     private[OptimizerCore] def fieldOriginalName(fieldName: FieldName): OriginalName =
-      recordFieldNames(fieldName).originalName
+      recordFields(fieldName).originalName
 
     override def equals(that: Any): Boolean = that match {
       case that: InlineableClassStructure =>
+        this.className == that.className &&
         this.allFields == that.allFields
       case _ =>
         false
     }
 
-    override def hashCode(): Int = allFields.##
+    override def hashCode(): Int = className.## ^ allFields.##
```

Comment:
Then again, if we actually need a `hashCode` (I could not find the usage site), we should have `equals` :-/